mlkem/cbd.c

```diff
@@ -54,21 +54,29 @@ static uint32_t load24_littleendian(const uint8_t x[3]) {
  *              - const uint8_t *buf: pointer to input byte array
  **************************************************/
 static void cbd2(poly *r, const uint8_t buf[2 * MLKEM_N / 4]) {
-  unsigned int i, j;
+  int i, j;
   uint32_t t, d;
   int16_t a, b;
 
-  for (i = 0; i < MLKEM_N / 8; i++) {
-    t = load32_littleendian(buf + 4 * i);
-    d = t & 0x55555555;
-    d += (t >> 1) & 0x55555555;
+  for (i = 0; i < MLKEM_N / 8; i++)  // clang-format off
+    ASSIGNS(i, j, a, b, t, d, OBJECT_WHOLE(r))
+    INVARIANT(i >= 0 && i <= MLKEM_N / 8)
+    INVARIANT(ARRAY_IN_BOUNDS(int, k, 0, (8 * i - 1), r->coeffs, -3, +3))  // clang-format on
```

Comment:
Aren't these bounds inclusive and should, thus, be -2 and +2? Or do I misremember things here?

Reply:
Great catch @mkannwischer. It is very confusing that `ARRAY_IN_BOUNDS` and `POLY_BOUND` etc. use different conventions. We should get that uniform at some point.

https://github.com/pq-code-package/mlkem-c-aarch64/issues/324